realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -907,14 +909,18 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeMaximumInt(
     if (!QUERY_VALID(env, pQuery) ||
         !COL_INDEX_AND_TYPE_VALID(env, pTable, columnIndex, type_Int) ||
         !ROW_INDEXES_VALID(env, pTable, start, end, limit))
-        return 0;
+        return NULL;
     try {
-        return pQuery->maximum_int(S(columnIndex), NULL, S(start), S(end), S(limit));
+        size_t ndx;
+        int64_t result = pQuery->maximum_int(S(columnIndex), NULL, S(start), S(end), S(limit), &ndx);
+        if (ndx != npos) {
```

Comment:
`npos` is a const in core `const std::size_t npos = std::size_t(-1);` in `array.hpp`